test/e2e/extensions.test.ts

```diff
@@ -8,12 +8,14 @@ function runTestExtensionTests() {
   test("should have access to VSCODE_PROXY_URI", async ({ codeServerPage }) => {
     const address = await getMaybeProxiedCodeServer(codeServerPage)
 
+    await codeServerPage.reloadUntilTestExtensionIsLoaded()
```

Comment:
Were you seeing the extension not get loaded at all on the first attempt or was it just taking a bit of time to load?

If the latter, my thinking is that we risk reloading the page multiple times if the extension host is a bit slow when really we just need to wait for the extension to get activated.  Something like:

```
this.codeServer.logger.debug("Waiting for test extension to load...")
await this.page.waitForSelector("text=test extension loaded")
```

If we do this you can ignore all my other comments as they will be obsolete.  :stuck_out_tongue: